mempool/cat/store.go

```diff
@@ -27,7 +27,7 @@ func (s *store) set(wtx *wrappedTx) bool {
 	}
 	s.mtx.Lock()
 	defer s.mtx.Unlock()
-	if tx, exists := s.txs[wtx.key]; !exists || tx.height == -1 {
+	if tx, exists := s.txs[wtx.key]; !exists || tx.isReserved() {
```

Comment:
[Question]Why should we store a reserved transaction if it will not be processed due to being a duplicate?

Reply:
idk that's a great question cc @cmwaters 

this was just adding a method for syntactic sugar, but it does seem like we should remove that check entirely